onmt/dynamic/build_vocab.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python
+from onmt.utils.logging import init_logger
+
+from onmt.dynamic.parse import DynamicArgumentParser
+from onmt.dynamic.opts import dynamic_preprocess_opts
+from onmt.dynamic.corpus import save_transformed_sample
+from onmt.dynamic.vocab import build_dynamic_fields, save_fields
+from onmt.dynamic.transform import make_transforms, save_transforms, \
+    get_specials, get_transforms_cls
+
+# from collections import defaultdict
+from collections import Counter
+
+def build_vocab_main(opts):
+    DynamicArgumentParser.validate_dynamic_corpus(opts)
+    DynamicArgumentParser.get_all_transform(opts)
+    init_logger()
+
+    transforms_cls = get_transforms_cls(opts._all_transform)
+    fields = None
+
+    transforms = make_transforms(opts, transforms_cls, fields)
+    counters_src, counters_tgt = save_transformed_sample(
+        opts, transforms, n_sample=25000, build_vocab=True)
```

Comment:
25k for `n_sample` seems not very enough to build vocab when corpus is very large and diversed. This could be a option maybe.

Reply:
In my test setup I had something like 20 or 30 datasets, hence the 25k per datasets. But it should absolutely be an option.